Doc/reference/datamodel.rst

```diff
@@ -2616,6 +2616,23 @@ are awaitable.
    :term:`awaitable` objects.  For instance, :class:`asyncio.Future` implements
    this method to be compatible with the :keyword:`await` expression.
 
+   .. note::
+
+      The language doesn't place any restrictions on the type or value of the
+      objects yielded by the iterator returned by `__await__`, as this is
+      specific to the implementation of the asynchronous execution framework
+      (e.g. :mod:`asyncio`) that will be managing the :term:`awaitable` object.
+      
```

Comment:
Trailing whitespace
```suggestion

```

(Oops, looks like I accidentally added this in my initial suggestion to split the paragraphs. Usually my editor catches this, but I did it via github UI).